src/main/java/seedu/address/logic/commands/CommandResult.java

```diff
@@ -13,104 +13,103 @@
-        this.feedbackToUser = requireNonNull(feedbackToUser);
-        this.showHelp = showHelp;
-        this.exit = exit;
+        this.cliFeedbackToUser = requireNonNull(cliFeedbackToUser);
+        this.shouldShowHelp = shouldShowHelp;
+        this.shouldExit = shouldExit;
         this.hasGuiInteraction = hasGuiInteraction;
-        this.selectedDoctor = Optional.ofNullable(selectedDoctor);
-        this.selectedPatient = Optional.ofNullable(selectedPatient);
+        this.selectedDoctor = selectedDoctor;
+        this.selectedPatient = selectedPatient;
     }
 
     /**
      * Constructs a {@code CommandResult} with the specified fields.
      */
-    public CommandResult(String feedbackToUser, boolean showHelp, boolean exit, boolean hasGuiInteraction) {
-        this(feedbackToUser, showHelp, exit, hasGuiInteraction, null, null);
+    public CommandResult(String cliFeedbackToUser, boolean shouldShowHelp,
+                         boolean shouldExit, boolean hasGuiInteraction) {
+        this(cliFeedbackToUser, shouldShowHelp, shouldExit,
+                hasGuiInteraction, null, null);
     }
 
     /**
      * Constructs a {@code CommandResult} with the specified arguments,
      * and {@code hasGuiInteraction} set to its default value.
      */
-    public CommandResult(String feedbackToUser, boolean showHelp, boolean exit) {
-        this(feedbackToUser, showHelp, exit, false);
+    public CommandResult(String cliFeedbackToUser, boolean shouldShowHelp, boolean shouldExit) {
+        this(cliFeedbackToUser, shouldShowHelp, shouldExit, false);
     }
 
     /**
      * Constructs a {@code CommandResult} with the specified {@code feedbackToUser},
      * and other fields set to their default value.
      */
-    public CommandResult(String feedbackToUser) {
-        this(feedbackToUser, false, false, false);
+    public CommandResult(String cliFeedbackToUser) {
+        this(cliFeedbackToUser, false, false, false);
     }
 
     /**
      * Constructs a {@code CommandResult} with selected doctor.
      */
-    public CommandResult(String feedbackToUser, boolean hasGuiInteraction, Doctor selectedDoctor) {
-        this(feedbackToUser, false, false, hasGuiInteraction, selectedDoctor, null);
+    public CommandResult(String cliFeedbackToUser,
+                         Doctor selectedDoctor) {
+        this(cliFeedbackToUser, false, false,
+                true, selectedDoctor, null);
     }
 
     /**
      * Constructs a {@code CommandResult} with selected patient.
      */
-    public CommandResult(String feedbackToUser, boolean hasGuiInteraction, Patient selectedPatient) {
-        this(feedbackToUser, false, false, hasGuiInteraction, null, selectedPatient);
+    public CommandResult(String cliFeedbackToUser,
+                         Patient selectedPatient) {
+        this(cliFeedbackToUser, false, false,
+                true, null, selectedPatient);
```

Comment:
Should we set defaults instead of calling the main constructor with boolean values?

Reply:
Hmm you are right! let me make the changes